google-home-community.groovy

```diff
@@ -562,6 +578,117 @@ private deviceTraitPreferences_Dock(deviceTrait) {
     }
 }
 
+@SuppressWarnings(['UnusedPrivateMethod', 'MethodSize'])
+private deviceTraitPreferences_EnergyStorage(deviceTrait) {
+    googleEnergyStorageDistanceUnitForUX = [
+        "KILOMETERS":      "Kilometers",
+        "MILES":           "Miles",
+    ]
+    googleCapacityUnits = [
+        "SECONDS":         "Seconds",
+        "MILES":           "Miles",
+        "KILOMETERS":      "Kilometers",
+        "PERCENTAGE":      "Percentage",
+        "KILOWATT_HOURS":  "Kilowatt Hours",
+    ]
+    section("Energy Storage Settings") {
+        input(
+            name: "${deviceTrait.name}.isRechargeable",
+            title: "Rechargeable",
+            type: "bool",
+            defaultValue: false,
+            required: true,
+            submitOnChange: true
+        )
+        input(
+            name: "${deviceTrait.name}.capacityRemainingRawValue",
+            title: "Capacity Remaining Value",
+            type: "text",
+            defaultValue: "battery",
+            required: true
+        )
+        input(
+            name: "${deviceTrait.name}.capacityRemainingUnit",
+            title: "Capacity Remaining Unit",
+            type: "enum",
+            options: googleCapacityUnits,
+            defaultValue: "PERCENTAGE",
+            multiple: false,
+            required: true,
+            submitOnChange: true
+        )
+    }
+    section(hideable: true, hidden: true, "Advanced Settings") {
+        input(
+            name: "${deviceTrait.name}.queryOnlyEnergyStorage",
+            title: "Query Only Energy Storage",
+            type: "bool",
+            defaultValue: true,
+            required: true,
+            submitOnChange: true
+        )
+        if (!deviceTrait.queryOnlyEnergyStorage) {
+            input(
+                name: "${deviceTrait.name}.chargeCommand",
+                title: "Charge Command",
+                type: "text",
+                required: true
+            )
+        }
```

Comment:
I noticed that this input exists when initially creating a new EnergyStorage trait, since the first time this page loads `deviceTrait.queryOnlyEnergyStorage` is `null`.  You can work around this by explicitly comparing to `false`:
```
if (deiviceTrait.queryOnlyEnergyStorage == false)
```

Probably a good idea to add a comment explaining that as well.

Reply:
Fixed that one, and one in the timer trait.